dpctl/_sycl_event.pyx

```diff
@@ -192,8 +193,20 @@ cdef class SyclEventRaw(_SyclEventRaw):
         """
         return self._event_ref
 
-    cpdef void wait(self):
-        DPCTLEvent_Wait(self._event_ref)
+    @staticmethod
+    cdef void _wait(SyclEventRaw event):
+        DPCTLEvent_WaitAndThrow(event._event_ref)
+
+    @staticmethod
+    def wait(event):
+        if isinstance(event, list):
```

Comment:
I think this should work even if `event` is a tuple, or any iterable object of events (need not be a list).